codeworld-api/src/CodeWorld/Picture.hs

```diff
@@ -51,6 +51,9 @@ data Picture
     = SolidPolygon CallStack
               [Point]
               !Bool
+    | SolidClosedCurve CallStack
+              [Point]
+              !Bool
```

Comment:
You should no longer need this `Bool`.  It only existed to distinguish between polygons and closed curves (i.e., smooth or not).  Since they are separate constructors, you no longer need to keep track.

Reply:
Tricky stuff. May need some help on this one :D I pushed the small edit up in the meantime (ie "S" to "s")

Reply:
Sure.  The way you've got it now, `SolidPolygon` always carries a value of `False` for the last field, and `SolidClosedCurve` always carries a value of `True`.  You can just remove the field from the type, and when you're pattern matching, you'll know whether it should be `True` or `False` based on which constructor it is.

Reply:
I removed the ```Bool``` constructors from ```SolidPolygon``` and ```SolidClosedCurve``` and then I get this. I think I know what to do with the first error, but the case statement looks like it's a bigger fix and I'm not sure if I'm on the right track seeing this.

![screenshot_2018-05-15_23-52-53](https://user-images.githubusercontent.com/8516378/40097372-a3b5c352-588a-11e8-9546-e848f00ada44.png)

Reply:
How do I resolve this type? I figured that ```s``` is ```smooth``` (so True or False) and was able to resolve the other parts by removing the ```smooth parameter``` in the case statements, but some guidance on this last part would be really appreciated :D 

![screenshot_2018-05-16_00-55-38](https://user-images.githubusercontent.com/8516378/40099084-5decc2c2-5893-11e8-91b1-e3e282318022.png)

Reply:
Sounds good